pytato/target/pycuda/numpy_like.py

```diff
@@ -0,0 +1,1105 @@
+        self.arg_names: Set[str] = set()
+        self.bound_arguments: Dict[str, DataInterface] = {}
+        
+    def fill_array_call(self, expr: IndexLambda, hlo: HighLevelOp) -> CgenMapperAccumulator:
+
+        array_name = self.vng('_pt_array')
+        memalloc_node_name = self.vng("_pt_memalloc")
+        # array_name would be passed into `add_memset_node`
+        memalloc_line = memalloc_call(array_name,
+                             memalloc_node_name,
+                             [],
+                             self.graph_name,
+                             nbytes= expr.size * expr.dtype.itemsize)
+        self.lines.append(memalloc_line)
+
+        memset_node_name = self.vng("_pt_memset")
+        memset_line = kernel_call(array_name = array_name,
+                                  node_name = memset_node_name,
+                                  dependencies = [memalloc_node_name],
+                                  graph_name = self.graph_name,
+                                  expr = expr,
+                                  hlo = hlo,
+                                  numpy_backend = self.numpy_backend)
+        self.lines.append(memset_line)
+        # The generated code should be:
+        # memset_node = g.add_memset_node([malloc_node_name], hlo.fill_value, array_name)
+        # (update self.lines)
+
+        return CgenMapperAccumulator(array_name, memset_node_name)
+
+
+    def map_index_lambda(self, expr: IndexLambda) -> CgenMapperAccumulator:
+        from pytato.raising import index_lambda_to_high_level_op
+        from pytato.raising import (FullOp, BinaryOp, WhereOp,
+                                    BroadcastOp, ReduceOp,
+                                    BinaryOpType)
+        hlo = index_lambda_to_high_level_op(expr)
+        lhs = self.vng("_pt_tmp")
+        rhs: ast.expr
+
+
+        # A+B
+        # rec_x1 = self.rec(hlo.x1)
+        # rec_x2 = self.rec(hlo.x2)
+
+        # array_name = self.vng("_pt_tmp")
+        # malloc_node_name = self.vng("_pt_memalloc")
+        # kernel_call_node_name = self.vng("_pt_kernel_call")
+        # line = memalloc_node(dependencies=[rec_x1.node_name, rec_x2.node_name],
+        #                      ...)
+
+        # kernel_call_node_name = g.add_kernel_node([memalloc_node_call],
+        #                                           element.axpy_kernel(),
+        #                                           rec_x1.buffer_name,
+        #                                           rec_x2.buffer_name,
+        #                                           array_name)
+        # self.lines.append(line)
+        # self.lines.append(knl_call_line)
+        # return CgenMapperAccumulator(array_name, kernel_call_node_name)
+
+        if isinstance(hlo, FullOp):
+            return self.fill_array_call(expr, hlo)
+             
+        elif isinstance(hlo, BinaryOp):
+            rec_x1 = self.rec(hlo.x1)
+            rec_x2 = self.rec(hlo.x2)
```

Comment:
Either raise an error if `not utils.are_shapes_equal(hlo.x1.shape, hlo.x2.shape)`, or,  generate loopy kernels for these as:
```python
>>> from pymbolic import var
>>> knl = lp.make_kernel(
...     "{[i, j]: 0<=i<100 and 0<=j<10}",
...     [lp.Assignment("out[i, j]", substitute(result.expr,
...                                            {"_0": var("i"), "_1": var("j")}))])
```